CMakeLists.txt

```diff
@@ -6,152 +6,46 @@ set(CMAKE_CXX_STANDARD 14)
-target_link_libraries(script_error_handling lua)
-add_executable(variadic_args examples/variadic_args.cpp)
-target_link_libraries(variadic_args lua)
-add_executable(coroutine examples/coroutine.cpp)
-target_link_libraries(coroutine lua)
-add_executable(calling_lua_functions examples/calling_lua_functions.cpp)
-target_link_libraries(calling_lua_functions lua)
-add_executable(customization examples/customization.cpp)
-target_link_libraries(customization lua)
-add_executable(usertype_var examples/usertype_var.cpp)
-target_link_libraries(usertype_var lua)
-add_executable(environments examples/environments.cpp)
-target_link_libraries(environments lua)
-add_executable(basic examples/basic.cpp)
-target_link_libraries(basic lua)
-add_executable(require examples/require.cpp)
-target_link_libraries(require lua)
-add_executable(usertype examples/usertype.cpp)
-target_link_libraries(usertype lua)
-add_executable(overloading examples/overloading.cpp)
-target_link_libraries(overloading lua)
-add_executable(usertype_automatic_operators examples/usertype_automatic_operators.cpp)
-target_link_libraries(usertype_automatic_operators lua)
-add_executable(containers_as_table examples/containers_as_table.cpp)
-target_link_libraries(containers_as_table lua)
-add_executable(usertype_special_functions examples/usertype_special_functions.cpp)
-target_link_libraries(usertype_special_functions lua)
-add_executable(stack_aligned_function examples/stack_aligned_function.cpp)
-target_link_libraries(stack_aligned_function lua)
-add_executable(optional_with_iteration examples/optional_with_iteration.cpp)
-target_link_libraries(optional_with_iteration lua)
-add_executable(config examples/config.cpp)
-target_link_libraries(config lua)
-add_executable(functions examples/functions.cpp)
-target_link_libraries(functions lua)
-install(TARGETS 
-        containers
-        tables
-        environments_on_functions
-        variables
-        usertype_bitfields
-        protected_functions
-        usertype_dynamic_getter_setter
-        environment_snooping
-        container_usertype_as_container
-        multi_results
-        self_call
-        any_return
-        usertype_advanced
-        usertype_initializers
-        namespacing
-        overloading_with_fallback
-        self_from_lua
-        usertype_simple
-        runtime_additions
-        usertype_call_from_c++
-        static_variables
-        script_error_handling
-        variadic_args
-        coroutine
-        calling_lua_functions
-        customization
-        usertype_var
-        environments
-        basic
-        require
-        usertype
-        overloading
-        usertype_automatic_operators
-        containers_as_table
-        usertype_special_functions
-        stack_aligned_function
-        optional_with_iteration
-        config
-        functions
-        RUNTIME DESTINATION bin/examples)
+set(LUA_VERSION "5.3" CACHE STRING "The version of Lua needed (5.1, 5.2, 5.3)")
```

Comment:
Do you need to handle Lua JIT? IIRC they are found the same way, but just unsure if more checks are needed.